sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/Exceptions.java

```diff
@@ -70,4 +70,8 @@ public static boolean isAvoidQuorumSelectionException(CosmosException cosmosExce
         return Exceptions.isStatusCode(cosmosException, HttpConstants.StatusCodes.GONE)
             && Exceptions.isSubStatusCode(cosmosException, HttpConstants.SubStatusCodes.LEASE_NOT_FOUND);
     }
+
+    public static boolean isClientAssignedSubStatusCode(int subStatusCode) {
+        return subStatusCode >= 20_000 && subStatusCode < 21_000;
```

Comment:
I recall there are more client assigned ones, like some in 10_000 range.

Reply:
Scoped it to 500s and 20* range.